client/views/admin/upgrade/UpgradePage/UpgradePage.tsx

```diff
@@ -0,0 +1,107 @@
+import { Throbber, Box } from '@rocket.chat/fuselage';
+import React, { ReactElement, useEffect, useRef, useState } from 'react';
+
+import type { UpgradeTabVariants } from '../../../../../lib/getUpgradeTabType';
+import Page from '../../../../components/Page';
+import PageHeader from '../../../../components/Page/PageHeader';
+import { useLayout } from '../../../../contexts/LayoutContext';
+import { useRouteParameter, useQueryStringParameter } from '../../../../contexts/RouterContext';
+import { useAbsoluteUrl } from '../../../../contexts/ServerContext';
+import { useLanguage } from '../../../../contexts/TranslationContext';
+import UpgradePageError from '../UpgradePageError';
+
+const iframeStyle = { width: '100%', height: '100%' };
+
+const urlMap = {
+	goFullyFeatured: 'https://go.rocket.chat/i/upgrade-ce-1-unregistered',
+	goFullyFeaturedRegistered: 'https://go.rocket.chat/i/upgrade-ce-1-registered',
+	trialGold: 'https://go.rocket.chat/i/upgrade-gold-trial',
+	trialEnterprise: 'https://go.rocket.chat/i/upgrade-ee-trial',
+	upgradeYourPlan: 'https://go.rocket.chat/i/upgrade-ce-2',
+};
+
+const getUrl = (type: UpgradeTabVariants, date: string | undefined, language: string): string => {
+	const urlParams = new URLSearchParams({ lang: language.toLowerCase() });
+	if (date) {
+		urlParams.set('date', date);
+	}
+
+	return `${urlMap[type]}?${urlParams.toString()}`;
+};
+
+type NavigationMessage = { goTo: string };
+
+const messageIsNavigation = (message: unknown): message is NavigationMessage => {
```

Comment:
Wording here is a little weird...

```suggestion
const isNavigationMessage = (message: unknown): message is NavigationMessage => {
```